src/Tools/SchemaTool.php

```diff
@@ -969,29 +971,84 @@ private function createSchemaForComparison(Schema $toSchema): Schema
         }
     }
 
-    /** @param string[] $primaryKeyColumns */
-    private function addPrimaryKeyConstraint(Table $table, array $primaryKeyColumns): void
+    private function getPrimaryKeyConstraint(Table $table): PrimaryKeyConstraint|Index|null
     {
-        if (class_exists(PrimaryKeyConstraint::class)) {
-            $primaryKeyColumnNames = [];
+        // DBAL < 4.3
+        if (! method_exists($table, 'getPrimaryKeyConstraint')) {
+            return $table->getPrimaryKey();
+        }
 
-            foreach ($primaryKeyColumns as $primaryKeyColumn) {
-                $primaryKeyColumnNames[] = new UnqualifiedName(Identifier::unquoted($primaryKeyColumn));
-            }
+        return $table->getPrimaryKeyConstraint();
+    }
 
-            $table->addPrimaryKeyConstraint(new PrimaryKeyConstraint(null, $primaryKeyColumnNames, true));
-        } else {
+    /** @param string[] $primaryKeyColumns */
```

Comment:
Should we use the same advanced types you used somewhere else in this MR? Maybe it will address the baselined issue.

```suggestion
    /** @param non-empty-list<non-empty-string> $primaryKeyColumns */
```